spec/lang/well-formed.md

```diff
@@ -341,6 +341,19 @@ impl ValueExpr {
                         }
                         Type::Int(left)
                     }
+                    IntWithOverflow(int_op) => {
+                        let Type::Int(int_ty) = left else {
+                            throw_ill_formed!("BinOp::IntWithOverflow: invalid left type");
+                        };
+                        ensure_wf(right == Type::Int(int_ty), "BinOp::IntWithOverflow: invalid right type")?;
+                        match int_op {
+                            IntBinOp::Add | IntBinOp::Sub | IntBinOp::Mul => {}
+                            _ => throw_ill_formed!(
+                                "BinOp::IntWithOverflow: op is not available as overflow variant"
+                            ),
```

Comment:
Instead of checking this dynamically, please use a separate enum so that it is impossible to even use a wrong operation here.

Reply:
Added the `IntBinOpWithOverflow`